### PR TITLE
Remove memory sanity checks

### PR DESCRIPTION
**Pull request by derobins**
Also simplifies the H5MM package, replacing most H5MM replacement functions with defines to C standard library functions

### CODE REVIEW

**Review by byrnHDF — APPROVED** [review]

File: release_docs/RELEASE.txt [hunk 1]
```diff
@@ -55,13 +55,29 @@ New Features
 
       (ADB - 2022/08/29, HDFFV-11329)
 
-    -
+    - Removed the --enable-memory-alloc-sanity-check Autotools configure option
```
Comment (byrnHDF) [inline release_docs/RELEASE.txt:58]: newer notes at top of section, right?

Comment (bmribler) [reply release_docs/RELEASE.txt:58]: That's what I thought...

File: src/H5.c [hunk 2]
```diff
@@ -1262,7 +1219,10 @@ H5resize_memory(void *mem, size_t size)
     FUNC_ENTER_API_NOINIT
     H5TRACE2("*x", "*xz", mem, size);
 
-    ret_value = H5MM_realloc(mem, size);
+    if (size != 0)
+        ret_value = H5MM_realloc(mem, size);
+    else if (mem)
+        ret_value = H5MM_xfree(mem);
```
Comment (bmribler) [inline src/H5.c:1225]: I read here https://www.tutorialspoint.com/c_standard_library/c_function_realloc.htm, it said
size − This is the new size for the memory block, in bytes. If it is 0 and ptr points to an existing block of memory, the memory block pointed by ptr is deallocated and a NULL pointer is returned.
Do we need to use H5MM_xfree instead?

File: src/H5MM.c [hunk 3]
```diff
@@ -449,122 +72,24 @@ H5MM_xstrdup(const char *s)
...
-    FUNC_LEAVE_NOAPI(ret_value)
-} /* end H5MM_strndup() */
-
 /*-------------------------------------------------------------------------
  * Function:    H5MM_xfree
  *
- * Purpose:     Just like free(3) except null pointers are allowed as
- *              arguments, and the return value (always NULL) can be
- *              assigned to the pointer whose memory was just freed:
+ * Purpose:     Just like free(3) except the return value (always NULL) can
```
Comment (fortnern) [inline src/H5MM.c:86]: Is H5MM_xfree still needed? Do any supported systems fail if we remove it (could test by #define H5MM_xfree H5MM_free)?

**Comment by fortnern** [comment]
It won't let me approve it for some reason but "approved"

**Review by fortnern — APPROVED** [review]

File: src/H5MM.c (hunk 3 above)
Comment (jhendersonHDF) [reply src/H5MM.c:86]: If it gets mapped to just plain `free` then I think we still have several places in the library that would do `mem = free` which won't compile, so we'd need to update those. But I don't think it's really needed.

**Comment by lrknox** [comment]
Test that failed was one of the dt_arith tests in the MacOS Clang CMake REL test configuration.
23: Testing hard normalized long double -> signed char conversions        
...
  22/2452 Test   #23: H5TEST-dt_arith ............................................................***Failed    0.24 sec

There were also 75+ "Failed ..." messages in the same test configuration, mostly in h5repack (and h5dump) tests, but all of those tests passed.

**Comment by jhendersonHDF** [comment]
> Test that failed was one of the dt_arith tests in the MacOS Clang CMake REL test configuration. 23: Testing hard normalized long double -> signed char conversions ... 22/2452 Test #23: H5TEST-dt_arith ............................................................***Failed 0.24 sec
> 
> There were also 75+ "Failed ..." messages in the same test configuration, mostly in h5repack (and h5dump) tests, but all of those tests passed.

This is the same test that fails in https://github.com/HDFGroup/hdf5/pull/1929. @derobins and I need to figure out what's going on there because it looks like we have a memory issue that's been hidden on MacOS for a long time.

File: src/H5MM.c (hunk 3 above)
Comment (fortnern) [reply src/H5MM.c:86]: We should be able to convert H5MM_xfree to a macro:
#define H5MM_xfree(A) (H5MM_free(A), NULL)

**Comment by derobins** [comment]
Need to investigate the long double conversion failures on MacOS

**Pull request closed without merging**